src/Accelerators/NNPA/Transform/ZLow/ZLowRewrite.cpp

```diff
@@ -0,0 +1,119 @@
+//===-------- ZLowRewrite.cpp - ZLow Rewrite Patterns ---------------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This pass implements optimizations for for ZLow operations.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Interfaces/ViewLikeInterface.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include "src/Accelerators/NNPA/Dialect/ZLow/ZLowOps.hpp"
+#include "src/Accelerators/NNPA/Pass/DLCPasses.hpp"
+
+using namespace mlir;
+
+/// This pattern rewrites
+/// ```mlir
+///   zlow.unstick(%input, %output)
+///   %view = viewOp(%output)
+///   zlow.stick(%view, %res)
+/// ```
+/// by removing `zlow.stick` and replacing `%res` by `%input`, which is
+/// constrained by that `%input` and `%res` have the same static shape.
+/// This pattern potentially removes `zlow.unstick` and `viewOp` if they are
+/// dangling.
+///
+/// `viewOp` can be any op that inherits ViewLikeOpInterface, e.g.
+/// memref.reinterpret_cast, memref.collapse_shape, memref.expand_shape.
+//
+class StickViewUnstickRemovalPattern : public OpRewritePattern<ZLowStickOp> {
```

Comment:
we should have a namespace for the conversion patterns of the ZLow and ZHigh dialects. Similar to what we now have for the krnl dialect.  Suggest to place ZLow in namespace `onnx_mlir::zlow`, similar for zhigh.

Reply:
I believe this issue is addressed by #1221 and #1222.